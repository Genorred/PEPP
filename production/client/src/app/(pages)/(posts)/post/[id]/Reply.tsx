import React from "react";
import {
  CommentTemplate,
  CommentTemplateI,
  UserCommentTemplateI
} from "@/app/(pages)/(posts)/post/[id]/CommentTemplate";
import { CornerDownRight } from "lucide-react";

const Reply = ({ comment, onFindRespondedComment }: {
  comment: CommentTemplateI & {
    parentId?: number;
    respondedComment?: {
      id: number
      user: UserCommentTemplateI
      message: string
    };
  };
  onFindRespondedComment: () => void;
}) => {
  return (
    <CommentTemplate comment={comment} className={"ml-4"}
                     replyFormProps={{
                       postId: comment.postId,
                       parentId: comment.parentId,
                       respondedCommentId: comment.id
                     }}
                     cardTop={comment.respondedComment && (
                       <div className="mb-2 p-2 bg-muted rounded-md text-sm"
                            onClick={onFindRespondedComment}>
                         <div className="flex items-center mb-1">
                           <CornerDownRight className="w-4 h-4 mr-2" />
                           <span className="font-semibold">{comment.respondedComment.user.username}</span>
                         </div>
                         <p className="text-muted-foreground line-clamp-2">{comment.respondedComment.message}</p>
                       </div>
                     )} />
  );
};

export default Reply;