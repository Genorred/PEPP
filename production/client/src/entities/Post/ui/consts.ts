export const variants = {
  comments: "Comments",
  reviews: "Reviews"
} as const;
export type Variants = typeof variants[keyof typeof variants];