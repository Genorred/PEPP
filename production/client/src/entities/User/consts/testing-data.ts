export const userInitialState = {
  id: 1,
  username: "j",
  email: "",
  createdAt: "",
  expireDate: 999999
};