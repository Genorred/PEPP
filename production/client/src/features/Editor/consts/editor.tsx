import { createPlateEditor, ParagraphPlugin, PlateLeaf } from "@udecode/plate-common/react";
import { BlockquotePlugin } from "@udecode/plate-block-quote/react";
import { HorizontalRulePlugin } from "@udecode/plate-horizontal-rule/react";
import { LinkPlugin } from "@udecode/plate-link/react";
import { LinkFloatingToolbar } from "@/entities/Post/ui/plate-ui/link-floating-toolbar";
import { ImagePlugin, MediaEmbedPlugin } from "@udecode/plate-media/react";
import { MentionInputPlugin, MentionPlugin } from "@udecode/plate-mention/react";
import { TableCellHeaderPlugin, TableCellPlugin, TablePlugin, TableRowPlugin } from "@udecode/plate-table/react";
import { TodoListPlugin } from "@udecode/plate-list/react";
import { ExcalidrawPlugin } from "@udecode/plate-excalidraw/react";
import { TogglePlugin } from "@udecode/plate-toggle/react";
import { ColumnItemPlugin, ColumnPlugin } from "@udecode/plate-layout/react";
import { HeadingPlugin } from "@udecode/plate-heading/react";
import { DatePlugin } from "@udecode/plate-date/react";
import {
  BoldPlugin,
  CodePlugin,
  ItalicPlugin,
  StrikethroughPlugin,
  SubscriptPlugin,
  SuperscriptPlugin,
  UnderlinePlugin
} from "@udecode/plate-basic-marks/react";
import { FontBackgroundColorPlugin, FontColorPlugin, FontSizePlugin } from "@udecode/plate-font/react";
import { HighlightPlugin } from "@udecode/plate-highlight/react";
import { KbdPlugin } from "@udecode/plate-kbd/react";
import { AlignPlugin } from "@udecode/plate-alignment/react";
import { IndentPlugin } from "@udecode/plate-indent/react";
import { IndentListPlugin } from "@udecode/plate-indent-list/react";
import { LineHeightPlugin } from "@udecode/plate-line-height/react";
import { CommentsPlugin } from "@udecode/plate-comments/react";
import { AutoformatPlugin } from "@udecode/plate-autoformat/react";
import { BlockMenuPlugin, BlockSelectionPlugin } from "@udecode/plate-selection/react";
import { CaptionPlugin } from "@udecode/plate-caption/react";
import { DndPlugin } from "@udecode/plate-dnd";
import { EmojiPlugin } from "@udecode/plate-emoji/react";
import { ExitBreakPlugin, SoftBreakPlugin } from "@udecode/plate-break/react";
import { NodeIdPlugin } from "@udecode/plate-node-id";
import { ResetNodePlugin } from "@udecode/plate-reset-node/react";
import { DeletePlugin } from "@udecode/plate-select";
import { SlashPlugin } from "@udecode/plate-slash-command/react";
import { DocxPlugin } from "@udecode/plate-docx";
import { CsvPlugin } from "@udecode/plate-csv";
import { MarkdownPlugin } from "@udecode/plate-markdown";
import { JuicePlugin } from "@udecode/plate-juice";
import { withDraggables } from "@/entities/Post/ui/plate-ui/with-draggables";
import { withPlaceholders } from "@/entities/Post/ui/plate-ui/placeholder";
import { BlockquoteElement } from "@/entities/Post/ui/plate-ui/blockquote-element";
import { ExcalidrawElement } from "@/entities/Post/ui/plate-ui/excalidraw-element";
import { HrElement } from "@/entities/Post/ui/plate-ui/hr-element";
import { ImageElement } from "@/entities/Post/ui/plate-ui/image-element";
import { LinkElement } from "@/entities/Post/ui/plate-ui/link-element";
import { ToggleElement } from "@/entities/Post/ui/plate-ui/toggle-element";
import { ColumnGroupElement } from "@/entities/Post/ui/plate-ui/column-group-element";
import { ColumnElement } from "@/entities/Post/ui/plate-ui/column-element";
import { HEADING_KEYS } from "@udecode/plate-heading";
import { withProps } from "@udecode/cn";
import { HeadingElement } from "@/entities/Post/ui/plate-ui/heading-element";
import { MediaEmbedElement } from "@/entities/Post/ui/plate-ui/media-embed-element";
import { MentionElement } from "@/entities/Post/ui/plate-ui/mention-element";
import { MentionInputElement } from "@/entities/Post/ui/plate-ui/mention-input-element";
import { ParagraphElement } from "@/entities/Post/ui/plate-ui/paragraph-element";
import { TableElement } from "@/entities/Post/ui/plate-ui/table-element";
import { TableRowElement } from "@/entities/Post/ui/plate-ui/table-row-element";
import { TableCellElement, TableCellHeaderElement } from "@/entities/Post/ui/plate-ui/table-cell-element";
import { TodoListElement } from "@/entities/Post/ui/plate-ui/todo-list-element";
import { DateElement } from "@/entities/Post/ui/plate-ui/date-element";
import { CodeLeaf } from "@/entities/Post/ui/plate-ui/code-leaf";
import { CommentLeaf } from "@/entities/Post/ui/plate-ui/comment-leaf";
import { HighlightLeaf } from "@/entities/Post/ui/plate-ui/highlight-leaf";
import { KbdLeaf } from "@/entities/Post/ui/plate-ui/kbd-leaf";
import React from "react";

export const editor = createPlateEditor({
  plugins: [
    ParagraphPlugin,
    BlockquotePlugin,
    HorizontalRulePlugin,
    LinkPlugin.configure({
      render: { afterEditable: () => <LinkFloatingToolbar /> }
    }),
    ImagePlugin,
    MediaEmbedPlugin,
    MentionPlugin,
    TablePlugin,
    TodoListPlugin,
    ExcalidrawPlugin,
    TogglePlugin,
    ColumnPlugin,
    HeadingPlugin,
    DatePlugin,
    BoldPlugin,
    ItalicPlugin,
    UnderlinePlugin,
    StrikethroughPlugin,
    CodePlugin,
    SubscriptPlugin,
    SuperscriptPlugin,
    FontColorPlugin,
    FontBackgroundColorPlugin,
    FontSizePlugin,
    HighlightPlugin,
    KbdPlugin,
    AlignPlugin.configure({
      inject: { targetPlugins: ["p", "h1", "h2", "h3"] }
    }),
    IndentPlugin.configure({
      inject: { targetPlugins: ["p", "h1", "h2", "h3"] }
    }),
    IndentListPlugin.configure({
      inject: { targetPlugins: ["p", "h1", "h2", "h3"] }
    }),
    LineHeightPlugin.configure({
      inject: {
        nodeProps: {
          defaultNodeValue: 1.5,
          validNodeValues: [1, 1.2, 1.5, 2, 3]
        },
        targetPlugins: ["p", "h1", "h2", "h3"]
      }
    }),
    CommentsPlugin,
    AutoformatPlugin.configure({
      options: {
        enableUndoOnDelete: true,
        rules: [
          // Usage: https://platejs.org/docs/autoformat
        ]
      }
    }),
    BlockSelectionPlugin,
    BlockMenuPlugin,
    CaptionPlugin.configure({
      options: { plugins: [ImagePlugin, MediaEmbedPlugin] }
    }),
    DndPlugin.configure({
      options: { enableScroller: true }
    }),
    EmojiPlugin,
    ExitBreakPlugin.configure({
      options: {
        rules: [
          {
            hotkey: "mod+enter"
          },
          {
            before: true,
            hotkey: "mod+shift+enter"
          },
          {
            hotkey: "enter",
            level: 1,
            query: {
              allow: ["h1", "h2", "h3"],
              end: true,
              start: true
            },
            relative: true
          }
        ]
      }
    }),
    NodeIdPlugin,
    ResetNodePlugin.configure({
      options: {
        rules: [
          // Usage: https://platejs.org/docs/reset-node
        ]
      }
    }),
    DeletePlugin,
    SoftBreakPlugin.configure({
      options: {
        rules: [
          { hotkey: "shift+enter" },
          {
            hotkey: "enter",
            query: {
              allow: ["code_block", "blockquote", "td", "th"]
            }
          }
        ]
      }
    }),
    SlashPlugin,
    DocxPlugin,
    CsvPlugin,
    MarkdownPlugin,
    JuicePlugin
  ],
  override: {
    components: withDraggables(withPlaceholders(({
      [BlockquotePlugin.key]: BlockquoteElement,
      [ExcalidrawPlugin.key]: ExcalidrawElement,
      [HorizontalRulePlugin.key]: HrElement,
      [ImagePlugin.key]: ImageElement,
      [LinkPlugin.key]: LinkElement,
      [TogglePlugin.key]: ToggleElement,
      [ColumnPlugin.key]: ColumnGroupElement,
      [ColumnItemPlugin.key]: ColumnElement,
      [HEADING_KEYS.h1]: withProps(HeadingElement, { variant: "h1" }),
      [HEADING_KEYS.h2]: withProps(HeadingElement, { variant: "h2" }),
      [HEADING_KEYS.h3]: withProps(HeadingElement, { variant: "h3" }),
      [HEADING_KEYS.h4]: withProps(HeadingElement, { variant: "h4" }),
      [HEADING_KEYS.h5]: withProps(HeadingElement, { variant: "h5" }),
      [HEADING_KEYS.h6]: withProps(HeadingElement, { variant: "h6" }),
      [MediaEmbedPlugin.key]: MediaEmbedElement,
      [MentionPlugin.key]: MentionElement,
      [MentionInputPlugin.key]: MentionInputElement,
      [ParagraphPlugin.key]: ParagraphElement,
      [TablePlugin.key]: TableElement,
      [TableRowPlugin.key]: TableRowElement,
      [TableCellPlugin.key]: TableCellElement,
      [TableCellHeaderPlugin.key]: TableCellHeaderElement,
      [TodoListPlugin.key]: TodoListElement,
      [DatePlugin.key]: DateElement,
      [BoldPlugin.key]: withProps(PlateLeaf, { as: "strong" }),
      [CodePlugin.key]: CodeLeaf,
      [CommentsPlugin.key]: CommentLeaf,
      [HighlightPlugin.key]: HighlightLeaf,
      [ItalicPlugin.key]: withProps(PlateLeaf, { as: "em" }),
      [KbdPlugin.key]: KbdLeaf,
      [StrikethroughPlugin.key]: withProps(PlateLeaf, { as: "s" }),
      [SubscriptPlugin.key]: withProps(PlateLeaf, { as: "sub" }),
      [SuperscriptPlugin.key]: withProps(PlateLeaf, { as: "sup" }),
      [UnderlinePlugin.key]: withProps(PlateLeaf, { as: "u" })
    })))
  },
  value: [
    {
      id: "1",
      type: "p",
      children: [{ text: "Hello, World!" }]
    }
  ]
});
export const { tf: editorTransformation } = editor;