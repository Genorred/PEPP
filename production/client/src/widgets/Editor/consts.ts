export const buttonNames = {
  save: "save",
  publish: "publish"
} as const;