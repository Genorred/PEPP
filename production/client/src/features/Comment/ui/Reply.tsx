import React, { useState } from "react";
import { CalendarIcon, CornerDownRight, MessageCircle } from "lucide-react";
import { Card, CardContent } from "@/shared/ui/card";
import Image from "next/image";
import { Button } from "@/shared/ui/button";
import CommentForm from "@/features/Comment/ui/CommentForm";
import { UserCommentTemplateI } from "@/features/Comment/model/model";

const Reply = ({ comment, onFindRespondedComment, onCreate }: {
  comment: {
    id: number;
    user: UserCommentTemplateI;

    createdAt: string;
    message: string;
    respondedComment?: {
      id: number
      user: UserCommentTemplateI
      message: string
    };
  };
  onFindRespondedComment?: () => void;
  onCreate: (respondedCommentId?: number) => (message: string, onError?: () => void) => void;
}) => {
  const state = useState(false);
  const [isReplying, setIsReplying] = state;
  return (
    <Card className={"w-full mb-4"}>
      {comment.respondedComment && (
        <div className="mb-2 p-2 bg-muted rounded-md text-sm"
             onClick={onFindRespondedComment}>
          <div className="flex items-center mb-1">
            <CornerDownRight className="w-4 h-4 mr-2" />
            <span className="font-semibold">{comment.respondedComment.user.username}</span>
          </div>
          <p className="text-muted-foreground line-clamp-2 break-words">{comment.respondedComment.message}</p>
        </div>
      )}
      <CardContent className="pt-4">
        <div className="flex items-center mb-4">
          {comment.user.img ? (
            <Image
              src={comment.user.img}
              alt={comment.user.username}
              width={40}
              height={40}
              className="rounded-full mr-3"
            />
          ) : (
            <div className="w-10 h-10 bg-gray-200 rounded-full mr-3" />
          )}
          <div>
            <h3 className="font-semibold">{comment.user.username}</h3>
            <p className="text-sm text-muted-foreground flex items-center">
              <CalendarIcon className="w-4 h-4 mr-1" />
              {comment.createdAt}
            </p>
          </div>
        </div>
        {/*<div className="flex items-center mb-2">*/}
        {/*  {[1, 2, 3, 4, 5].map((star) => (*/}
        {/*    <Star*/}
        {/*      key={star}*/}
        {/*      className={`w-5 h-5 ${*/}
        {/*        star <= comment.rating ? "text-yellow-400 fill-yellow-400" : "text-gray-300"*/}
        {/*      }`}*/}
        {/*    />*/}
        {/*  ))}*/}
        {/*</div>*/}
        <p className="mb-4">{comment.message}</p>
        <div className="flex items-center space-x-2">
          <Button variant="outline" size="sm" onClick={() => setIsReplying(!isReplying)}>
            <MessageCircle className="w-4 h-4 mr-2" />
            {isReplying ? "Cancel" : "Reply"}
          </Button>
        </div>
        {isReplying &&
          <CommentForm onCreate={onCreate(comment.id)} isReplyingState={state} placeholder={"reply"} />}
      </CardContent>
    </Card>
  );
};

export default Reply;