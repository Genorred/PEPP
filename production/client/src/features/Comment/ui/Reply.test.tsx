import { fireEvent, screen } from "@testing-library/react";
import { renderWithProviders } from "@/shared/utils/test-utils";
import { userSlice } from "@/entities/User/model/user.slice";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { userInitialState } from "@/entities/User/consts/testing-data";
import Reply from "./Reply";

const commentData = {
  id: 99,
  message: "",
  user: {
    id: 0,
    username: ""
  },
  createdAt: "08/07/2025"
};
const respondedComment = {
  id: 1,
  message: "",
  user: {
    id: 2,
    username: "",
    img: ""
  }
};
const funcClosure = jest.fn();
const onCreate = jest.fn().mockImplementation((id?: number) => funcClosure);
describe("Reply", () => {
  it("should call create function", async () => {
    const { asFragment } = renderWithProviders(<Reply onCreate={onCreate} comment={commentData} />, {
      slices: [userSlice],
      preloadedState: {
        user: {
          user: userInitialState
        }
      }
    });

    await userEvent.click(screen.getByTestId("reply-button"));
    expect(onCreate).toHaveBeenCalledWith(commentData.id);

    const textarea = screen.getByTestId("comment-form");
    const message = "Satisfying";
    await userEvent.type(textarea, message);
    await userEvent.click(screen.getByTestId("comment-form-button-submit"));
    expect(funcClosure).toHaveBeenCalledWith(message, expect.any(Function));

    expect(asFragment()).toMatchSnapshot();
  });
  it("should display provided responded comment", () => {
    const { asFragment } = renderWithProviders(
      <Reply onCreate={onCreate}
             comment={{ ...commentData, respondedComment }}
      />
    );
    expect(screen.getByTestId("responded-comment")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it("should not display unprovided responded comment", () => {
    const { asFragment } = renderWithProviders(
      <Reply onCreate={onCreate}
             comment={commentData}
      />
    );
    expect(screen.queryByTestId("responded-comment")).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });
});