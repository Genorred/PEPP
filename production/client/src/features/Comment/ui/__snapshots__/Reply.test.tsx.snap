// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Comment should open created replies 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full mb-4"
  >
    <div
      class="p-6 pt-4"
    >
      <div
        class="flex items-center mb-4"
      >
        <a
          class="flex items-center gap-2"
          href="/profile/0"
        >
          <span
            class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full"
          >
            <span
              class="flex h-full w-full items-center justify-center rounded-full bg-muted"
            />
          </span>
          <div>
            <p
              class="text-sm font-medium"
            />
          </div>
        </a>
      </div>
      <p
        class="mb-4 break-words"
      />
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          data-testid="reply-button"
        >
          <svg
            class="lucide lucide-message-circle w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
          </svg>
          Cancel
        </button>
        <button
          aria-controls="replies-0"
          aria-expanded="true"
          class="justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3 flex items-center"
          data-testid="open-replies-button"
        >
          <svg
            class="lucide lucide-chevron-down w-4 h-4 mr-2 transition-transform duration-200 rotate-180"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
          Hide Replies (1)
        </button>
      </div>
      <form
        class="mt-4"
      >
        <textarea
          class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px] mb-2"
          data-testid="comment-form"
          placeholder="Write your reply..."
        />
        <div
          class="mt-4 flex justify-end space-x-2"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
            data-testid="comment-form-button-cancel"
            type="button"
          >
            Cancel
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3"
            data-testid="comment-form-button-submit"
          >
            Submit Message
          </button>
        </div>
      </form>
    </div>
    <div
      class="p-6 flex flex-col items-start pt-0"
      data-testid="replies"
    >
      <div
        class="w-full transition-all duration-300 ease-in-out max-h-full opacity-100"
        id="replies-0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Reply should call create function 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full mb-4"
  >
    <div
      class="p-6 pt-4"
    >
      <div
        class="flex items-center mb-4"
      >
        <div
          class="w-10 h-10 bg-gray-200 rounded-full mr-3"
        />
        <div>
          <h3
            class="font-semibold"
          />
          <p
            class="text-sm text-muted-foreground flex items-center"
          >
            <svg
              class="lucide lucide-calendar w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            08/07/2025
          </p>
        </div>
      </div>
      <p
        class="mb-4"
      />
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          data-testid="reply-button"
        >
          <svg
            class="lucide lucide-message-circle w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
          </svg>
          Cancel
        </button>
      </div>
      <form
        class="mt-4"
      >
        <textarea
          class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px] mb-2"
          data-testid="comment-form"
          placeholder="Write your reply..."
        />
        <div
          class="mt-4 flex justify-end space-x-2"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
            data-testid="comment-form-button-cancel"
            type="button"
          >
            Cancel
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3"
            data-testid="comment-form-button-submit"
          >
            Submit Message
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Reply should display provided responded comment 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full mb-4"
  >
    <div
      class="mb-2 p-2 bg-muted rounded-md text-sm"
      data-testid="responded-comment"
    >
      <div
        class="flex items-center mb-1"
      >
        <svg
          class="lucide lucide-corner-down-right w-4 h-4 mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 10 20 15 15 20"
          />
          <path
            d="M4 4v7a4 4 0 0 0 4 4h12"
          />
        </svg>
        <span
          class="font-semibold"
        />
      </div>
      <p
        class="text-muted-foreground line-clamp-2 break-words"
      />
    </div>
    <div
      class="p-6 pt-4"
    >
      <div
        class="flex items-center mb-4"
      >
        <div
          class="w-10 h-10 bg-gray-200 rounded-full mr-3"
        />
        <div>
          <h3
            class="font-semibold"
          />
          <p
            class="text-sm text-muted-foreground flex items-center"
          >
            <svg
              class="lucide lucide-calendar w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            08/07/2025
          </p>
        </div>
      </div>
      <p
        class="mb-4"
      />
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          data-testid="reply-button"
        >
          <svg
            class="lucide lucide-message-circle w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
          </svg>
          Reply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Reply should not display unprovided responded comment 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full mb-4"
  >
    <div
      class="p-6 pt-4"
    >
      <div
        class="flex items-center mb-4"
      >
        <div
          class="w-10 h-10 bg-gray-200 rounded-full mr-3"
        />
        <div>
          <h3
            class="font-semibold"
          />
          <p
            class="text-sm text-muted-foreground flex items-center"
          >
            <svg
              class="lucide lucide-calendar w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            08/07/2025
          </p>
        </div>
      </div>
      <p
        class="mb-4"
      />
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          data-testid="reply-button"
        >
          <svg
            class="lucide lucide-message-circle w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
          </svg>
          Reply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Reply should open created replies 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full mb-4"
  >
    <div
      class="p-6 pt-4"
    >
      <div
        class="flex items-center mb-4"
      >
        <div
          class="w-10 h-10 bg-gray-200 rounded-full mr-3"
        />
        <div>
          <h3
            class="font-semibold"
          />
          <p
            class="text-sm text-muted-foreground flex items-center"
          >
            <svg
              class="lucide lucide-calendar w-4 h-4 mr-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
            </svg>
            08/07/2025
          </p>
        </div>
      </div>
      <p
        class="mb-4"
      />
      <div
        class="flex items-center space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          data-testid="reply-button"
        >
          <svg
            class="lucide lucide-message-circle w-4 h-4 mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
          </svg>
          Cancel
        </button>
      </div>
      <form
        class="mt-4"
      >
        <textarea
          class="flex w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm min-h-[100px] mb-2"
          data-testid="comment-form"
          placeholder="Write your reply..."
        />
        <div
          class="mt-4 flex justify-end space-x-2"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
            data-testid="comment-form-button-cancel"
            type="button"
          >
            Cancel
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-9 rounded-md px-3"
            data-testid="comment-form-button-submit"
          >
            Submit Message
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
